portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,12 +40,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    ArrayList<Pair<String, String>> messages = new ArrayList<Pair<String, String>>();
+    ArrayList<Triplet<String, String, Long>> messages = new ArrayList<Triplet<String, String, Long>>();
```

Comment:
In java a pretty good rule of thumb when building data structures is to ask how much semantic meaning is implicit vs. explicit.

In this case, messages' type requires a lot of context about how the data is stored. A better approach is to create a struct (either a class or a proto) that encapsulates this information, and then you make the implicit behavior of the stored data explicit.

You don't have to do that here if you don't want to, but we'll require it on the full project.

If you choose to encapsulate this type out into a class, you could write it in this class and it would start like:

private static class Message {
  final String name;
  final String text;
  final long id;
  
  Message(String name, String text, long Id) {
    ...
  }
}

Reply:
I wont change it because Id have to track down code in the js file to fix. However I'll keep it in mind for further reference